### PR TITLE
 ath79: add support for Adtran Bluesocket BSAP-192x board

### PR DESCRIPTION
**Pull request by bgonyer**
ath79: add support for Adtran Bluesocket BSAP-192x board

This board is an Atheros AR9344 based dual band enterprise access point
known as Adtran models BSAP-1920 and BSAP-1925. The BSAP-192x series
are the same hardware, BSAP-1925 antennae are external and detachable
RP-SMA, while the BSAP-1920 are internal. They have an internal 3V TTL
serial port header that is populated with pins, connection info is
115200, 8N1. 
Board is similar to PowerCloud Systems CAP324 except larger flash size.

Specification:
- System-On-Chip: Atheros AR9344
- CPU/Speed: 560 MHz
- Flash-Chip: Spansion FL256SAIFRO
- Flash size: 32 MiB
- RAM: 128 MiB
- phy0: Atheros AR9340 5GHz 802.11an
- phy1: Atheros AR9300 2.4GHz 802.11bgn
- PHY: Atheros AR8035-A
- Power: PoE 802.3af and 12V DC at 2A

Installation:
1. Connect to the serial console using a TTL 3V adapter only
   115200 bps, 8 data bits, no parity, 1 stop bit
   Pin 1 (VCC) is marked with an arrow. Use RX TX GND
	   Pin 	4 	3 	2 	1
	Signal 	RX 	TX 	GND VCC
Due to a u-boot quirk you may have to connect serial cables after the AP boots 
or you will be sitting at a screen that never changes.

2. Interrupt the bootloader using its password, which is: b100:
3. at bootloader 192x > prompt enter the following commands
4. setenv serverip <your_tftp_server_ip>
5. setenv ipaddr <valid_ip_range_>
6. setenv mfg_state
7. saveenv
8. tftpboot 0x3000000 openwrt-ath79-generic-adtran_bsap1925-initramfs-kernel.bin
9. bootm 0x3000000
10. Wait for the status LED to go solid green
11. perform a sysupgrade using openwrt-ath79-generic-adtran_bsap1925-squashfs-sysupgrade.bin
12. after this point the bootloader password will be r00t

### CODE REVIEW

File: target/linux/ath79/config-5.15 [hunk 1]
```diff
@@ -34,15 +34,16 @@ CONFIG_CPU_R4K_CACHE_TLB=y
...
 CONFIG_CPU_SUPPORTS_MSA=y
-CONFIG_CRYPTO_BLAKE2S=y
 CONFIG_CRYPTO_LIB_BLAKE2S_GENERIC=y
 CONFIG_CRYPTO_LIB_POLY1305_RSIZE=2
 CONFIG_CRYPTO_RNG2=y
 CONFIG_CSRC_R4K=y
 CONFIG_DMA_NONCOHERENT=y
 CONFIG_DTC=y
 CONFIG_EARLY_PRINTK=y
+CONFIG_ETHERNET_PACKET_MANGLE=y
```
Comment (hauke) [inline target/linux/ath79/config-5.15:44]: Are these changes really needed for this board?
If they are not needed just leave them out. If they are needed please add them into a separate commit and describe why they are needed.

File: target/linux/ath79/dts/ar9344_adtran_bsap192x.dtsi [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	status = "okay";
+
+	ath9k: wifi@0,0 {
+		compatible = "pci168c,0030";
+		reg = <0x0000 0 0 0 0>;
+		nvmem-cells = <&macaddr_senao_0>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <(1)>;
+		mtd-cal-data = <&art 0x5000>;
+		qca,no-eeprom;
```
Comment (hauke) [inline target/linux/ath79/dts/ar9344_adtran_bsap192x.dtsi:88]: Is `qca,no-eeprom` still needed when you provide the eeprom over mtd?

File: target/linux/ath79/image/generic.mk [hunk 3]
```diff
@@ -246,6 +246,32 @@ define Device/adtran_bsap1840
...
+  DEVICE_VENDOR := Adtran
+  DEVICE_PACKAGES := uboot-envtools
+  IMAGE_SIZE := 13312k
+# TODO: Define IMAGE/sysupgrade.bin
+
+# TODO: Define KERNEL
+# e.g. KERNEL := kernel-bin | append-dtb | lzma | pad-to $$(BLOCKSIZE)
+
+# TODO: (maybe?) Define KERNEL_INITRAMFS
+# e.g. KERNEL_INITRAMFS := kernel-bin | append-dtb
```
Comment (hauke) [inline target/linux/ath79/image/generic.mk:260]: Please remove these todos

**Comment by hauke** [comment]
Please line break the commit message after 72 characters

File: target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom [hunk 4]
```diff
@@ -82,6 +82,10 @@ case "$FIRMWARE" in
 	;;
 "ath9k-eeprom-pci-0000:00:00.0.bin")
 	case $board in
+	adtran,bsap1920|\
+	adtran,bsap1925)
+		caldata_extract "art" 0x5000 0x440
+    ;;
```
Comment (hauke) [inline target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom:88]: indention is wrong here

Comment (bgonyer) [reply target/linux/ath79/generic/base-files/etc/hotplug.d/firmware/10-ath9k-eeprom]: fixed indention

**Comment by hauke** [comment]
The follow up commits are all just fixing problems in the initial commit, please squash them all together and force push them into this pull request. While at it please add line breaks after 72 characters into the commit description of the initial commit.
